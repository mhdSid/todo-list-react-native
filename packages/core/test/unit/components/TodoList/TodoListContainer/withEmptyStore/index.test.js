import { render } from '@testing-library/react-native'
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store'
import Component from '../../../../../../src/components/TodoList/TodoListContainer'
import selectors from '../../../../../lib/selector/todoList/todoListContainer'
import addTodoButtonSelectors from '../../../../../lib/selector/button/addTodoButton'
import virtualizedTodoListSelectors from '../../../../../lib/selector/todoList/virtualizedTodoList'
import MockThemeProvider from '../../../../../lib/mocks/MockThemeProvider'

jest.mock('expo-local-authentication', () => ({
  authenticateAsync: jest.fn(() => ({}))
}))

const mockStore = configureStore([thunk])

const store = mockStore({
  todoList: {
    list: []
  },
  auth: {
    user: {
      isLoggedIn: false
    }
  },
  loading: {
    loaders: {}
  },
  theme: {
    theme: 'light'
  }
})

describe('TodoList', () => {
  test('root exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const root = await element.findByTestId(selectors.root)
    expect(root).toBeDefined()
  })
  test('virtualizedList exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const virtualizedList = await element.findByTestId(virtualizedTodoListSelectors.root)
    expect(virtualizedList).toBeDefined()
  })
  test('addTodoButton exists', async () => {
    const element = render(MockThemeProvider(Component, store))
    const addTodoButton = await element.findByTestId(addTodoButtonSelectors.root)
    expect(addTodoButton).toBeDefined()
  })
  test('snapshot', () => {
    const data = render(MockThemeProvider(Component, store)).toJSON()
    expect(data).toMatchSnapshot()
  })
})
