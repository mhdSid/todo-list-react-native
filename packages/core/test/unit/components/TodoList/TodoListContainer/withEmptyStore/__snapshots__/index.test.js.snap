// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList snapshot 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
  testID="todoList.root"
>
  <RCTScrollView
    ListEmptyComponent={<EmptyTodoList />}
    contentContainerStyle={
      Object {
        "paddingBottom": 20,
      }
    }
    contentInsetAdjustmentBehavior="automatic"
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={0.2}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    showsVerticalScrollIndicator={true}
    stickyHeaderIndices={Array []}
    testID="virtualizedTodoList.root"
    windowSize={5}
  >
    <RCTRefreshControl />
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginTop": 50,
          }
        }
        testID="emptyTodoList.root"
      >
        <Text
          style={
            Array [
              Object {
                "marginBottom": 4,
              },
              Object {
                "color": "#333",
              },
            ]
          }
          testID="emptyTodoList.title"
        >
          Looks like your TODO list is empty!
        </Text>
        <Text
          style={
            Array [
              Object {
                "marginBottom": 4,
              },
              Object {
                "color": "#333",
              },
            ]
          }
          testID="emptyTodoList.subtitle"
        >
          Add a new task to the list :)
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityValue={
      Object {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 75,
          "bottom": 30,
          "height": 75,
          "justifyContent": "center",
          "position": "absolute",
          "right": 20,
          "textAlign": "center",
          "width": 75,
          "zIndex": 100,
        },
        Object {
          "backgroundColor": "black",
        },
      ]
    }
    testID="addTodoButton.root"
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 50,
            "marginBottom": 5,
          },
          Object {
            "color": "white",
          },
        ]
      }
      testID="addTodoButton.text"
    >
      +
    </Text>
  </View>
</View>
`;
